File: Java/Memory.java
package W2.Latihan1.Java;

class Memory extends Hardware {
    private int frequency;
    private int memorySize;
    private boolean supportsCuda;

    public Memory(int frequency, int memorySize, boolean supportsCuda) {
        this.setFrequency(frequency);
        this.setMemorySize(memorySize);
        this.setSupportCuda(supportsCuda);
    }

    public Memory() {
    }

    public int getFrequency() {
        return frequency;
    }

    public void setFrequency(int frequency) {
        this.frequency = frequency;
    }

    public int getMemorySize() {
        return memorySize;
    }

    public void setMemorySize(int memorySize) {
        this.memorySize = memorySize;
    }

    public boolean getSupportCuda() {
        return supportsCuda;
    }

    public void setSupportCuda(boolean supportsCuda) {
        this.supportsCuda = supportsCuda;
    }

    public void printMemory() {
        System.out.println("Frequency    : " + this.frequency);
        System.out.println("Memory Size  : " + this.memorySize);
        System.out.println("Support CUDA : " + this.supportsCuda);
    }

}

File: Python/Memory.py
from Hardware import Hardware


class Memory(Hardware):
    __Frequency = ""
    __MemorySize = 0
    __CUDA = ""
    
    def __init__(self, Frequency = "", MemorySize = 0, CUDA = ""):
        self.__Frequency = Frequency
        self.__MemorySize = MemorySize
        self.__CUDA = CUDA
        
    def setFrequency(self, Frequency):
        self.__Frequency = Frequency
        
    def getFrequency(self):
        return self.__Frequency
    
    def setMemory(self, MemorySize):
        self.__MemorySize = MemorySize
        
    def getMemory(self):
        return self.__MemorySize
    
    def setCUDA(self, CUDA):
        self.__CUDA = CUDA
        
    def getCUDA(self):
        return self.__CUDA


    def printMemory(self):
        print("Frequency        : " + str(self.__Frequency))
        print("Memory Size      : " + str(self.__MemorySize))
        print("Support CUDA     : " + str(self.__CUDA))
        